lib/runner/request-helpers-presend.js

```diff
@@ -48,13 +50,31 @@ module.exports = [
             authInterface,
             authSignHook = function () {
                 try {
-                    authHandler.sign(authInterface, context.item.request, function (err) { done(err); });
+                    authHandler.sign(authInterface, context.item.request, function (err) {
```

Comment:
In the future, `authHandler` should be adjusted to never throw, so that `try`-`catch` can be avoided.

Reply:
We might not be able to do it. In the future, this part is going to be out of runtime, and we can't trust it. So, we'll have to retain it forever.

Reply:
Alright, we'll have to add a `finally` block so that optimization bailout does not occur.